Add install script

Pysis868 opened the pull request:
Developed to not only include setting-up the expected configuration
file in its location, but also add the program to an expected location
on the PATH for many terminal users.

Not a huge use case, but thought I would start to provide the
convenience, and we can always improve on it later.

Pysis868 committed:
Add install script

Developed to not only include setting-up the expected configuration
file in its location, but also add the program to an expected location
on the PATH for many terminal users.

Not a huge use case, but thought I would start to provide the
convenience, and we can always improve on it later.

MattSurabian commented:
I was actually thinking about this after reading this article and the comments thread: http://www.linux-magazine.com/Online/Blogs/Productivity-Sauce/Connecting-to-proXPN-VPN-on-Linux-with-the-proXPN-Bash-Client

Pysis868 committed:
Update install.sh with minor change

Fixed-up the syntax for the rest of the variables to be more consistent.

Pysis868 committed:
Add ability to optionally install login credentials

Added functionality to install script to allow the user to choose whether they also want to install system-wide login credentials as another step int he install script.

Pysis868 committed:
Add command line arguments to install script

These enable basic help text for the first main command line option for 'forcing' overwriting existing system data the program relies on for whatever reason.

Pysis868 commented:
So, would we be able to merge this?

MattSurabian commented:
Definitely! I just haven't had a chance to try it out yet. I'll do my best to make some time tonight.

Pysis868 commented:
Ah, ok, just making sure there were no comments to add to it.  Thanks!

MattSurabian commented:
Probably not, I looked it over quickly the other day and nothing jumped out. Giving it another look now.

MattSurabian commented on install.sh:
```diff
@@ -0,0 +1,121 @@
+  if [[ -f "$executableSystemInstallPath" ]]; then
+    rm "$executableSystemInstallPath";
+  fi
+  if ln -s "$(readlink -f "$executableFilePath")" "$executableSystemInstallPath"; then
```
I get the benefit of using a symlink and updating via git but I'm not sure everyone trying to use this will realize they need to keep their initial download around or the program will stop functioning.

Maybe there should be an option for symlinking but the default behavior is a straight copy?

Pysis868 replied:
Understandable.

I will add a flag just in case this functionality is wanted, and use copy as the default method instead.

MattSurabian replied:
This symlink behavior is definitely wanted by me and will be how I use this script ;-) thanks! I know it adds some more logical checks which is a bummer but I think it'll be easier for most users.

MattSurabian commented on install.sh:
```diff
@@ -0,0 +1,121 @@
+  echo "System-wide credentials file already exists; continuing installation...";
+else
+  while [[ "$storeCredentialsChoice" != "y" && "$storeCredentialsChoice" != "n" ]]; do
+    read -p "Do you want to enter and store your credentials? (y/n): " storeCredentialsChoice;
```
This is great :+1:

MattSurabian commented on README.md:
```diff
@@ -62,7 +62,10 @@ brew install openvpn
+To use this script clone this repo or download the source from the releases page. The recommendation is to create a `proxpn` directory in `etc` and store the included `proxpn.ovpn` file there.
+The main script should be given execute permissions `chmod +x proxpn` and copied to somewhere in your `$PATH`. The recommendation is to copy the script to `/usr/local/bin` or `/usr/bin`. This should allow you to run the command `proxpn` from any location.
+
+This can now all be done using the provided installation script: `./install.sh`  
```
I'd adjust this section to start by recommending the install script and then saying something like, "if you'd prefer to install manually..." cause the install script is legit and folks should totally use it.

Pysis868 replied:
Will do.

MattSurabian commented:
Left a few small comments. This looks really great thanks! If it's not too much trouble, squashing down to a single commit would be awesome, but it's not a big deal either way.

MattSurabian commented:
Now that #1 is done, I think I'll cut another release as soon as this install script is ready to go. :100:

MattSurabian commented:
What do you think about changing the config paths to `/usr/local/etc`

MattSurabian commented:
Oops clicked the wrong button! Opening a new issue about possibly changing the install and config paths but that doesn't need to hold this up.

Pysis868 committed:
Add alternative installation method of the main executable as a copy as the new default instead, while still allowing making a symlink.

Pysis868 committed:
Merge commit '101655f6911d8c2e19e0b78e01fecbb584514e00' into install-script

Pysis868 committed:
Build on merged doc updates by revising installation section.

Using the script should now be more obvious, as well as the steps it performs, if the user wants to manually recreate the process themself.
Also added quotes for most paths, fixed some typos, code indentation, message spacing, and other misc changes.

Pysis868 commented:
Try those!

I had a small idea and was going to do some merging, not only with the master updates that I already included, but also the code from #17 & #18 that I was going to build off of, but instead of trying to figure out that part of GitHub and wondering if that's stepping out of bounds here, I'll just let you continue those tasks.

MattSurabian commented:
Looks good thanks! :+1:
